api/server/router/network/filter.go

```diff
@@ -29,6 +29,29 @@ func filterNetworkByType(nws []types.NetworkResource, netType string) ([]types.N
 	return retNws, nil
 }
 
+func filterNetworkByUse(nws []types.NetworkResource, danglingOnly bool) ([]types.NetworkResource, error) {
+	retNws := []types.NetworkResource{}
+
+	switch danglingOnly {
```

Comment:
But how will this work when `dangling=false`. In that case we'd have to append non-dangling networks to retNws. Which is done only when, either, the network has been attached to more than a container or is a pre-defined network.